crates/matrix-sdk-crypto/src/machine/mod.rs

```diff
@@ -1109,7 +1109,11 @@ impl OlmMachine {
         decrypted: &mut OlmDecryptionInfo,
         changes: &mut Changes,
     ) -> OlmResult<()> {
-        debug!("Received a decrypted to-device event");
+        debug!(
+            sender_device_keys =
+                ?decrypted.result.event.sender_device_keys().map(|k| (k.curve25519_key(), k.ed25519_key())).unwrap_or((None, None)),
```

Comment:
nit: you can probably [`Option::unzip`](https://doc.rust-lang.org/std/option/enum.Option.html#method.unzip) somewhere around here

Maybe:
```suggestion
                ?decrypted.result.event.sender_device_keys().map(|k| (k.curve25519_key(), k.ed25519_key())).unzip(),
```

Reply:
Erm, not sure that works.

The problem is that `k.curve25519_key()` and `k.ed25519_key()` both return an Option, so the result of my `map` is  `Option<(Option<_>, Option<_>)>`. Assuming the keys are populated, when I `unzip` that, I get `(Some(Some(k1)), Some(Some(k2))`.

Reply:
I see, let's keep it like this then!

Reply:
Probably meant:

```rust
decrypted.result.event
    .sender_device_keys()
    .and_then(|k| k.curve25519_key().zip(k.ed25519_key()))
```

Reply:
> Probably meant:
> 
> ```rust
> decrypted.result.event
>     .sender_device_keys()
>     .and_then(|k| k.curve25519_key().zip(k.ed25519_key()))
> ```

well, that might be nice, but AFAICT it doesn't handle having one key and not the other? So it would either be `Some(k1, k2)` or `None`?

I guess it's very unlikely to have just one key, but I think I'd rather allow for it